File: Java/ContainsDuplicate2.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class ContainsDuplicate2 {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> seen = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int val = nums[i];
            if (seen.containsKey(val) && i - seen.get(val) <= k) {
                return true;
            }
            seen.put(val, i);
        }

        return false;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        System.out.print("Enter number of elements: ");
        int n = sc.nextInt();
        int[] nums = new int[n];
        System.out.println("Enter the elements:");
        for (int i = 0; i < n; i++) {
            nums[i] = sc.nextInt();
        }

        System.out.print("Enter k (distance): ");
        int k = sc.nextInt();

        ContainsDuplicate2 obj = new ContainsDuplicate2();
        boolean result = obj.containsNearbyDuplicate(nums, k);

        if (result) {
            System.out.println("Duplicate within distance " + k + " exists.");
        } else {
            System.out.println("No duplicate within distance " + k + ".");
        }

        sc.close();
    }
}

File: LeetCode/Python/ContainsDuplicate2.py
from typing import List
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        seen = {}

        for i, val in enumerate(nums):
            if val in seen and i - seen[val] <= k:
                return True
            else:
                seen[val] = i
        
        return False
    
s=Solution()
ls=list(map(int,input("Enter elements : ").split()))
k=int(input("k :"))
if s.containsNearbyDuplicate(ls,k):
    print("True")
else:
    print("False")
